translator/source_code/combine_wzdx.py

```diff
@@ -0,0 +1,299 @@
+            cotrip_file, 'r').read())
+    except ValueError as e:
+        raise ValueError(
+            'One or more files specified are invalid. Please specify valid geojson files!') from None
+
+    combined_wzdx = find_duplicate_features_and_combine(
+        wzdx_icone, wzdx_cotrip)
+
+    if combined_wzdx:
+        with open(outputfile, 'w+') as f:
+            f.write(json.dumps(combined_wzdx, indent=2))
+        print(f'Combined WZDx message was written to {outputfile}.')
+    else:
+        print('No duplicate WZDx messages were found. Output file was not created')
+
+
+def find_duplicate_features_and_combine(wzdx_source: dict, wzdx_destination: dict) -> dict:
+    """Find duplicate features in WZDx messages and combine source duplicates into destination message
+
+    Args:
+        wzdx_source: Source WZDx message
+        wzdx_destination: Destination WZDx message
+
+    Returns: 
+        Destination WZDx message, updated from source if new data was found
+    """
+
+    # Make copy of destination object to combine duplicates into
+    combined_wzdx = copy.deepcopy(wzdx_destination)
+
+    # Iterate over destination features, for each search for duplicate source features. If any are found, combine them into the
+    # combined_wzdx feature
+    for index, destination_feature in enumerate(wzdx_destination["features"]):
+
+        # Generate polygon from destination feature
+        polygon = generate_buffer_polygon_from_linestring(
+            destination_feature['geometry']['coordinates'], polygon_width_meters)
+
+        # Get list of duplicate source features
+        iconeIndexes = iterate_feature(polygon, wzdx_source)
+
+        # Iteratively combine them with the combined feature
+        for iconeIndex in iconeIndexes:
+            combined_wzdx = combine_wzdx(
+                combined_wzdx, index, wzdx_source, iconeIndex)
+
+    # If features are identical, then no duplicates were found, return None. Else return combined message
+    if combined_wzdx == wzdx_destination:
+        return None
+    else:
+        return combined_wzdx
+
+
+def combine_wzdx(wzdx_destination: dict, destination_feature_index: int, wzdx_source: dict, source_feature_index: int) -> dict:
+    """Combine WZDx messages given duplicate feature indexes
+
+    Args:
+        combined_wzdx: Destination WZDx message (dict)
+        combined_feature_index: Index of destination feature to combine (int)
+        wzdx_icone: Destination WZDx message (dict)
+        icone_feature_index: Index of source feature to combine (int)
+
+    Returns: 
+        Destination WZDx message, updated from source if new data was found
+    """
+
+    # If indexes do not exist for either source or destination, immediately return destination object
+    if len(wzdx_destination['features']) <= destination_feature_index or len(wzdx_source['features']) <= source_feature_index:
+        destination_feature = wzdx_destination['features'][destination_feature_index]
```

Comment:
Can we remove? This would blow up if the destination_feature_index was greater than the length and we aren't using this variable anywhere. 
```suggestion
```

Reply:
Completely agree, not sure how that made it through, my mistake

Reply:
@payneBrandon I removed it.